src/MeltFraction/MeltingParameterization.jl

```diff
@@ -89,6 +90,80 @@ function show(io::IO, g::MeltingParam_Caricchi)
 end
 #-------------------------------------------------------------------------
 
+# Melnik  -------------------------------------------------------
+"""
+    MeltingParam_Melnik()
+
+Implements the T-dependent melting parameterisation used by Melnik and coworkers
+```math
+    x = {  T \\over 1000.0}
+```
+```math
+    \\theta = { a + b * x + c * x^2 + d * x^3}
+```
+```math
+    \\phi_{melt} = {1.0 \\over (1.0 + e^\\theta)}
+```
+Note that T is in Kelvin. 
+
+As default parameters we employ:
+```math
+a=517.9,  b=-1619.0, c=1699.0, d = -597.4
+```
+which gives a reasonable fit to experimental data for basalt.
+
+Data for rhyolite are:
+```math
+a=3043.0,  b=-10552.0, c=12204.9, d = -4709.0 
+```
+
+![MeltingParam_Melnik](./assets/img/MeltingParam_Melnik.png)
+Red: Rhyolite, Blue: Basalt
+
+References
+====
+"""
+@with_kw_noshow struct MeltingParam_Melnik{T,U,U1} <: AbstractMeltingParam{T}
+    a::GeoUnit{T,U} =  517.9NoUnits
+    b::GeoUnit{T,U} = -1619.0NoUnits
+    c::GeoUnit{T,U} = 1699.0NoUnits
+    d::GeoUnit{T,U} = -597.4NoUnits
+    Tchar::GeoUnit{T,U1} = 1000K # normalization
+    apply_bounds::Bool = true
+end
+MeltingParam_Melnik(args...) = MeltingParam_Melnik(convert.(GeoUnit, args)...)
+
+function param_info(s::MeltingParam_Melnik) # info about the struct
+    return MaterialParamsInfo(;
+        Equation=L"\phi = f(T)"
+    )
+end
+
+# Calculation routines
+function (p::MeltingParam_Melnik)(; T, kwargs...)
+    @unpack_val a, b, c, d, Tchar = p
+    x = T / Tchar
+    θ = a + b * x + c * x^2 + d * x^3;
+    ϕ = inv(1.0 + exp(θ))
+    return ϕ
+end
+
+function compute_dϕdT(p::MeltingParam_Melnik; T, kwargs...)
+    @unpack_val a, b, c, d, Tchar = p
+    
+    x = T / Tchar
+
+    dϕdT = -((2T*c) / (Tchar^2) + b / Tchar + (3d*((x)^2)) / Tchar)*(1 / ((1.0 + exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3)))^2))*exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3))
```

Comment:
```suggestion
    dϕdT = -((2T*c) / (Tchar^2) + b / Tchar + (3d*((x)^2)) / Tchar)*(1 / ((1.0 + exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3)))^2))*exp(a + (T*b) / Tchar + c*((x)^2) + d*((x)^3))
```
May be worth to pre-compute some repeated values here; may help also with readability. And many brackets can be removed.

Reply:
good point